code/Makefile

```diff
@@ -191,8 +194,16 @@ graphs: $(GRAPH_PACKAGES)
 ###
 
 analysis: check_stack
+	- rm -rf $(ANALYSIS_FOLDER)
 	@mkdir -p $(ANALYSIS_FOLDER)
 	cd $(SCRIPT_FOLDER) && stack exec -- runghc DataTableGen.hs
+	cd $(SCRIPT_FOLDER) && stack exec -- runghc TypeGraphGen.hs
+	stack install dotgen graphmod
```

Comment:
typically, it's best to do the `stack install` first because if it fails, there's no point doing any of the rest.